### PR TITLE
Add backwards-compatible validation for RawModuleDefV8.

### PR DESCRIPTION
**Pull request by kazimuth**
# Description of Changes

Plus unit and integration tests.
I'm still working through errors in the integration tests -- I think I'm going to need to something slightly more clever in a few places -- but generally this is ready for review.

# API and ABI breaking changes

N/A

# Expected complexity level and risk

This is tricky because we're adding a new code path that's supposed to replicate the behavior of another complex code path, but I think once we add the other end (ModuleDef -> TableSchema) comparing the results will be enough to validate that stuff works.

# Testing

I test on both synthetic v8 module definitions and actual module definitions from the repo.

### CODE REVIEW

File: crates/schema/src/def/validate/v8.rs [hunk 1]
```diff
@@ -1,9 +1,716 @@
...
+    #[test]
+    fn not_nominal() {
+        let inner_not_nominal_type = AlgebraicType::product([("b", AlgebraicType::U32)]);
+        let invalid_type = AlgebraicType::product([("a", inner_not_nominal_type.clone())]);
+        let mut builder = RawModuleDefV8Builder::default();
+        builder.add_type_for_tests("Invalid", invalid_type.clone());
+        builder.add_reducer_for_tests("silly", ProductType::from([("a", invalid_type.clone())]));
+        let result: Result<ModuleDef> = builder.finish().try_into();
+
+        expect_error_matching!(result, ValidationError::NotNominalNormalForm { location, ty } => {
```
Comment (kazimuth) [inline crates/schema/src/def/validate/v8.rs:648]: Note that these names will be changed once #1590 is merged

**Comment by kazimuth** [comment]
@Centril @cloutiertyler I consider this ready for merge.

File: crates/schema/src/def/validate/v8.rs [hunk 2]
```diff
@@ -1,9 +1,761 @@
...
+    let RawConstraintDefV8 {
+        constraint_name,
+        constraints,
+        columns,
+    } = constraint;
+
+    if constraints.has_primary_key() {
+        if let Some(col) = columns.as_singleton() {
+            if primary_key.is_none() {
+                *primary_key = Some(col);
```
Comment (cloutiertyler) [inline crates/schema/src/def/validate/v8.rs:260]: What's the point of this. Seems unused? Or am I misunderstanding? Is this mutating the argument?

Comment (kazimuth) [reply crates/schema/src/def/validate/v8.rs:260]: Yeah, I'll rewrite this with comments.

**Review by cloutiertyler — APPROVED** [review]
In principle this all seems fine to me.

File: crates/schema/src/def/validate/v9.rs [hunk 3]
```diff
@@ -755,11 +769,73 @@ enum TypeDefOrUse {
     Use,
 }
 
+fn check_table_names_consistent_with_type_names(
```
Comment (cloutiertyler) [inline crates/schema/src/def/validate/v9.rs:772]: Why do we have this requirement? Can I not have this:

```rust
#[table(name="logged_in_user")]
#[table(name="logged_out_user")]
struct User {
    ...
}
```

Comment (kazimuth) [reply crates/schema/src/def/validate/v9.rs:772]: oh you're completely right, i was just pondering if I actually needed this. I will discard.

**Pull request merged**